__tests__/script.test.js

```diff
@@ -8,13 +8,20 @@ describe('Test suite for the SDK', () => {
   const xhrMock = {
     open: jest.fn(),
     setRequestHeader: jest.fn(),
+    onload: jest.fn(),
     onreadystatechange: jest.fn(),
-    send: jest.fn(),
     readyState: 4,
-    responseText: JSON.stringify({}),
+    responseText: JSON.stringify({
+      source: {
+        config: {},
+        destinations: [],
+      },
+    }),
     status: 200,
   };
 
+  xhrMock['send'] = jest.fn(() => xhrMock.onload());
```

Comment:
Why is this needed?

Reply:
Our source config fetching call uses the `xhr.onload` approach. Previously we didn't bother to check this because xhr.send getting triggered was enough. But now we have to capture source config call and other API calls.